pkg/operator/operator.go

```diff
@@ -964,6 +979,97 @@ func (o *Operator) loadUserWorkloadConfig(ctx context.Context) (*manifests.UserW
 	return manifests.NewUserWorkloadConfigFromConfigMap(userCM)
 }
 
+func (o *Operator) mergeClusterMonitoringCRD(ctx context.Context, c *manifests.Config) error {
+	crd, err := o.client.GetClusterMonitoring(ctx, clusterResourceName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get ClusterMonitoring CRD: %w", err)
+	}
+
+	if crd.Spec.AlertmanagerConfig.DeploymentMode != "" {
```

Comment:
DeploymentMode can't be empty given the API validatoin?

Reply:
You're right, the API validation ensures a default value. I'll remove this check.